src/StreamJsonRpc/Reflection/MessageFormatterProgressTracker.cs

```diff
@@ -0,0 +1,239 @@
+                }
+            }
+
+            return iProgressOfTType;
+        }
+
+        /// <summary>
+        /// Saves a <see cref="ProgressParamInformation"/> instance obtained from the given <see cref="object"/> into <see cref="progressMap"/>.
+        /// </summary>
+        /// <param name="value">The object which should implement <see cref="IProgress{T}"/> to create the <see cref="ProgressParamInformation"/> instance.</param>
+        /// <returns>The assigned <see cref="long"/> progres ID.</returns>
+        public long AddProgressObjectToMap(object value)
+        {
+            Requires.NotNull(value, nameof(value));
+
+            if (this.RequestIdBeingSerialized == null)
+            {
+                throw new NotSupportedException(Resources.ProgressObjectInResponseOrNotificationError);
+            }
+
+            lock (this.progressLock)
+            {
+                long progressId = this.NextProgressId++;
+                this.requestProgressMap.Add(this.RequestIdBeingSerialized.Value, progressId);
+
+                this.progressMap.Add(progressId, new ProgressParamInformation(value));
+
+                return progressId;
+            }
+        }
+
+        /// <summary>
+        /// Clears the objects in <see cref="requestProgressMap"/> and <see cref="progressMap"/> associated to the given request ID.
+        /// </summary>
+        /// <param name="requestId">The id of the request whose associated object need to be cleared.</param>
+        public void ClearProgressMaps(long requestId)
+        {
+            lock (this.progressLock)
+            {
+                object progressId;
+
+                if (this.requestProgressMap.TryGetValue(requestId, out progressId))
+                {
+                    this.requestProgressMap.Remove(requestId);
+                    this.progressMap.Remove(progressId);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the <see cref="ProgressParamInformation"/> object associated with the given progress id.
+        /// </summary>
+        /// <param name="progressId">The key to obtain the <see cref="ProgressParamInformation"/> object from <see cref="progressMap"/>.</param>
+        /// <param name="valueType">Output parameter to store the obtained <see cref="ProgressParamInformation"/> object.</param>
+        public bool GetProgressTypeToReport(object progressId, out ProgressParamInformation valueType)
+        {
+            lock (this.progressLock)
+            {
+                if (this.progressMap.TryGetValue(progressId, out ProgressParamInformation progressInfo))
+                {
+                    valueType = progressInfo;
+                    return true;
+                }
+
+                valueType = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="JsonProgress{T}"/> class.
```

Comment:
>JsonProgress [](start = 53, length = 12)

Scrub mention of the `JsonProgress<T>` class from xml doc comments on your public members. This class is no longer visible to them, and it's not info they need to know. What they need to know is that it is an `IProgress<T>` instance for use on the receiving end of an RPC call.